src/utils.ts

```diff
@@ -1,15 +1,54 @@
 import { SelectionEntry, UIDisplayData } from '@aics/simularium-viewer';
 import { VisibilitySelectionMap } from './constants';
 
-export const getNewSelectionMap = (
+export const getNewMapAfterTopLevelCheckboxClick = (
   agentName: string,
   currentVisibilityMap: VisibilitySelectionMap
 ): VisibilitySelectionMap => {
   const newMap: VisibilitySelectionMap = { ...currentVisibilityMap };
-  if (currentVisibilityMap[agentName].length === 0) {
+  if (currentVisibilityMap[agentName].includes(agentName)) {
+    newMap[agentName] = [];
+  } else {
     newMap[agentName] = [agentName];
+  }
+  return newMap;
+};
+
+export const getNewMapAfterDisplayStateClick = (
+  agentName: string,
+  displayStateName: string,
+  allDisplayStates: string[],
+  currentVisibilityMap: VisibilitySelectionMap
```

Comment:
If we're not able to refactor or rename it, would really appreciate a comment on `VisibilitySelectionMap` that describes how to parse it at the very least!

I didn't really understand it until you talked me through it, and there isn't really any documentation for it in this codebase.
> if visMap[agentName] = [agentName] ... the agent is not selected (Default)
> if visMap[agentName] = [ ] ... the agent is selected and by extension any display states are also selected
> if visMap[agentName] = [ displayStateName1, displayStateName2... ] .. some but not all display states are selected, the agent checkbox is by definition indeterminate

Reply:
Where do you think would be the most useful place for this comment, next to the type definition?

Reply:
Next to the type definition makes sense to me, since that was the first place I checked.